src/main/kotlin/org/koitharu/kotatsu/parsers/site/manga18/en/Hentai3zCc.kt

```diff
@@ -0,0 +1,96 @@
+package org.koitharu.kotatsu.parsers.site.madara.en
+
+
+import org.koitharu.kotatsu.parsers.MangaLoaderContext
+import org.koitharu.kotatsu.parsers.MangaSourceParser
+import org.koitharu.kotatsu.parsers.model.Manga
+import org.koitharu.kotatsu.parsers.model.MangaSource
+import org.koitharu.kotatsu.parsers.model.MangaTag
+import org.koitharu.kotatsu.parsers.model.RATING_UNKNOWN
+import org.koitharu.kotatsu.parsers.model.SortOrder
+import org.koitharu.kotatsu.parsers.site.madara.Manga18Parser
+import org.koitharu.kotatsu.parsers.util.attrAsRelativeUrlOrNull
+import org.koitharu.kotatsu.parsers.util.domain
+import org.koitharu.kotatsu.parsers.util.generateUid
+import org.koitharu.kotatsu.parsers.util.host
+import org.koitharu.kotatsu.parsers.util.parseFailed
+import org.koitharu.kotatsu.parsers.util.parseHtml
+import org.koitharu.kotatsu.parsers.util.selectFirstOrThrow
+import org.koitharu.kotatsu.parsers.util.toAbsoluteUrl
+import org.koitharu.kotatsu.parsers.util.urlEncoded
+
+
+@MangaSourceParser("HENTAI3ZCC", "Hentai3z Cc", "en")
+internal class Hentai3zCc(context: MangaLoaderContext) :
+	Manga18Parser(context, MangaSource.HENTAI3ZCC, "hentai3z.cc") {
+
+	override val isNsfwSource = true
+
+	override suspend fun getListPage(
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder,
+	): List<Manga> {
+		val url = buildString {
+			append("https://")
+			append(domain)
+			val pages = page + 1
+			when {
+				!query.isNullOrEmpty() -> {
+					append("/$listeurl/")
+					append(pages.toString())
+					append("?search=")
+					append(query.urlEncoded())
+					append("&")
+				}
+
+				!tags.isNullOrEmpty() -> {
+					append("/$tagUrl/")
+					for (tag in tags) {
+						append(tag.key)
+					}
+					append("/")
+					append(pages.toString())
+					append("?")
+				}
+
+				else -> {
+					append("/$listeurl/")
+					append(pages.toString())
+					append("?")
+				}
+			}
+			append("order_by=")
+			when (sortOrder) {
+				SortOrder.POPULARITY -> append("views")
+				SortOrder.UPDATED -> append("lastest")
+				SortOrder.ALPHABETICAL -> append("name")
+				else -> append("latest")
+			}
+		}
+		val doc = webClient.httpGet(url).parseHtml()
+
+
+		return doc.select("div.story_item").map { div ->
+			val href = div.selectFirst("a")?.attrAsRelativeUrlOrNull("href") ?: div.parseFailed("Link not found")
```

Comment:
```suggestion
			val href = div.selectFirstOrThrow("a").attrAsRelativeUrl("href")
```

Why? In this case we will get more informative error message